webtest/forms.py

```diff
@@ -616,11 +633,12 @@ def submit(self, name=None, index=None, value=None, **args):
 
         """
         fields = self.submit_fields(name, index=index, submit_value=value)
```

Comment:
Retrieving the `submit_field` first, and then passing to `submit_fields` would probably be a bit nicer (i.e. avoid calling `_get_submit_field` twice), although would require a change of signature